Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -59,13 +61,17 @@ void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
     voltage_ = batteryData.voltage;
     current_ = batteryData.current;
 
-    hour_= batteryData.time.hour()+batteryData.time.minute()/60.0+batteryData.time.second()/3600.0+batteryData.time.msec()/3600000.0;
+    hour_= batteryData.time.hour()
+            +batteryData.time.minute()/HOUR_TO_M
+            +batteryData.time.second()/S_TO_HOURS
+            +batteryData.time.msec()/MS_TO_HOURS;
```

Comment:
there should be an easier way to do this, where you utilize QTime objects, and that way you'll only need to measure in one unit (such as Msec) then do one simple conversion to hours, rendering the other constants useless. Read up on this documentation if you aren't sure how to proceed: http://doc.qt.io/qt-5/qtime.html